servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultHost.java

```diff
@@ -143,6 +150,14 @@ public void markClosed() {
             final C cConn = (C) conn;
             cConn.closeAsyncGracefully().subscribe();
         }
+        if (oldState.state != CLOSED_CONN_STATE) {
+            // this is the first time this was marked closed so we need to let the observer know.
+            // TODO: do we need to signal closed for this host? Why isn't it closed? The
+            //  closing should probably be re-worked to funnel closing behavior through one place
+            //  and also define what being closed means: just the host isn't used anymore for new
+            //  requests/connections or does it also mean that all connections have closed?
```

Comment:
This looks like a bug but not one that I want to address here. I think there will be a bit more massaging in `DefaultHost` now that we have `SequentialExecutor` and can probably extract the outlier detector and let it synchronously read host state so long as it's within the executor.